### PR TITLE
azurerm_cognitive_account: deprecate sku in favour of sku_name

### PR DESCRIPTION
**Pull request by katbyte**
partially addresses #1500

### CODE REVIEW

File: azurerm/internal/services/cognitive/tests/resource_arm_cognitive_account_test.go [hunk 1]
```diff
@@ -202,6 +225,24 @@ resource "azurerm_resource_group" "test" {
   location = "%s"
 }
 
+resource "azurerm_cognitive_account" "test" {
+  name                = "acctestcogacc-%d"
```
Comment (tracypholmes) [inline azurerm/internal/services/cognitive/tests/resource_arm_cognitive_account_test.go]: Would we not change `"acctestcogacc-%d"` to `"acctestCOGacc-%d" or `"acctestCOGACC-%d"`?

File: website/docs/guides/2.0-upgrade-guide.html.markdown [hunk 2]
```diff
@@ -231,6 +231,10 @@ The AzureAD Data Sources and Resources have been moved to [the new AzureAD Provi
 
 A guide on how to migrate to using the new Provider [can be found here](https://www.terraform.io/docs/providers/azurerm/guides/migrating-to-azuread.html).
 
+### Resource: `azurerm_cognitive_account`
+
+The deprecated `sku` block has been simplified and replaced by the `sku_name` property and will be removed.
```
Comment (tracypholmes) [inline website/docs/guides/2.0-upgrade-guide.html.markdown]:
```suggestion
The deprecated `sku` block has been replaced by the `sku_name` field and will be removed.

The `sku_name` field will become Required.
```

File: azurerm/internal/services/cognitive/resource_arm_cognitive_account.go [hunk 3]
```diff
@@ -84,10 +84,23 @@ func resourceArmCognitiveAccount() *schema.Resource {
 				}, false),
 			},
 
+			"sku_name": {
+				Type:          schema.TypeString,
+				Optional:      true, // required in 2.0
+				Computed:      true, // remove in 2.0
+				ConflictsWith: []string{"sku"},
+				ValidateFunc: validation.StringInSlice([]string{
+					"F0", "F1", "S0", "S1", "S2", "S3", "S4", "S5", "S6", "P0", "P1", "P2",
```
Comment (WodansSon) [inline azurerm/internal/services/cognitive/resource_arm_cognitive_account.go]: Should these be separated with an underscore to be consistent with the other `sku_name`s?
```suggestion
					"F_0", "F_1", "S_0", "S_1", "S_2", "S_3", "S_4", "S_5", "S_6", "P_0", "P_1", "P_2",
```

Comment (katbyte) [reply azurerm/internal/services/cognitive/resource_arm_cognitive_account.go]: I don't want to make confusing alternatives, its up to azure to change it

File: azurerm/internal/services/cognitive/tests/resource_arm_cognitive_account_test.go (hunk 1 above)
Comment (katbyte) [reply azurerm/internal/services/cognitive/tests/resource_arm_cognitive_account_test.go]: Thats something to tackle in a different PR

**Pull request merged**